bee-bundle/src/transaction.rs

```diff
@@ -366,24 +384,53 @@ impl Transaction {
     pub fn from_trits(buffer: &Trits<impl RawEncoding<Trit = Btrit> + ?Sized>) -> Result<Self, TransactionError> {
         let trits = buffer.encode::<T1B1Buf>();
 
+        let value =
+            i64::try_from(trits[VALUE.trit_offset.start..VALUE.trit_offset.start + VALUE.trit_offset.length].to_buf())?;
+
+        let timestamp = i64::try_from(
+            trits[TIMESTAMP.trit_offset.start..TIMESTAMP.trit_offset.start + TIMESTAMP.trit_offset.length].to_buf(),
+        )? as u64;
+        let index =
+            i64::try_from(trits[INDEX.trit_offset.start..INDEX.trit_offset.start + INDEX.trit_offset.length].to_buf())?
+                as usize;
+        let last_index = i64::try_from(
+            trits[LAST_INDEX.trit_offset.start..LAST_INDEX.trit_offset.start + LAST_INDEX.trit_offset.length].to_buf(),
+        )? as usize;
+
+        let attachment_ts = i64::try_from(
+            trits[ATTACHMENT_TS.trit_offset.start..ATTACHMENT_TS.trit_offset.start + ATTACHMENT_TS.trit_offset.length]
+                .to_buf(),
+        )? as u64;
+
+        let attachment_lbts = i64::try_from(
+            trits[ATTACHMENT_LBTS.trit_offset.start
+                ..ATTACHMENT_LBTS.trit_offset.start + ATTACHMENT_LBTS.trit_offset.length]
+                .to_buf(),
+        )? as u64;
+        let attachment_ubts = i64::try_from(
+            trits[ATTACHMENT_UBTS.trit_offset.start
+                ..ATTACHMENT_UBTS.trit_offset.start + ATTACHMENT_UBTS.trit_offset.length]
+                .to_buf(),
+        )? as u64;
+
         let transaction = Self::builder()
             .with_payload(Payload(
                 trits[PAYLOAD.trit_offset.start..PAYLOAD.trit_offset.start + PAYLOAD.trit_offset.length].to_buf(),
             ))
             .with_address(Address(
                 trits[ADDRESS.trit_offset.start..ADDRESS.trit_offset.start + ADDRESS.trit_offset.length].to_buf(),
             ))
-            .with_value(Value::from_inner_unchecked(0))
+            .with_value(Value::from_inner_unchecked(value))
             .with_obsolete_tag(Tag(trits[OBSOLETE_TAG.trit_offset.start
                 ..OBSOLETE_TAG.trit_offset.start + OBSOLETE_TAG.trit_offset.length]
                 .to_buf()))
-            .with_timestamp(Timestamp::from_inner_unchecked(0))
-            .with_index(Index::from_inner_unchecked(0))
-            .with_last_index(Index::from_inner_unchecked(0))
+            .with_timestamp(Timestamp::from_inner_unchecked(timestamp))
+            .with_index(Index::from_inner_unchecked(index))
+            .with_last_index(Index::from_inner_unchecked(last_index))
             .with_tag(Tag(trits
                 [TAG.trit_offset.start..TAG.trit_offset.start + TAG.trit_offset.length]
                 .to_buf()))
-            .with_attachment_ts(Timestamp(0))
+            .with_attachment_ts(Timestamp(attachment_ts))
```

Comment:
Should this use `from_inner_unchecked` as well ?

Reply:
yes